lib/drives/util/recurrence_options_indicator.dart

```diff
@@ -0,0 +1,122 @@
+    final List<DateTime> afterDays = after.getAllInstances(start: start.toUtc());
+    final List<DateTime> addedDays = afterDays.where((DateTime day) => !beforeDays.contains(day)).toList();
+    final List<DateTime> removedDays = beforeDays.where((DateTime day) => !afterDays.contains(day)).toList();
+    final List<DateTime> allDays = (beforeDays + afterDays).toSet().toList()
+      ..sort((DateTime a, DateTime b) => a.compareTo(b));
+
+    if (allDays.isEmpty) {
+      return Text(S.of(context).pageRecurringDriveDetailUpcomingDrivesEmpty);
+    }
+
+    final List<Widget> days = allDays
+        .map<Widget>(
+          (DateTime elem) => RecurrenceOptionsIndicatorDay(
+            day: elem,
+            type: addedDays.contains(elem)
+                ? RecurrenceOptionsIndicatorType.added
+                : removedDays.contains(elem)
+                    ? RecurrenceOptionsIndicatorType.removed
+                    : RecurrenceOptionsIndicatorType.constant,
+          ),
+        )
+        .toList();
+
+    final int maxDaysFront =
+        max(4, allDays.where((DateTime elem) => elem.difference(start) < const Duration(days: 14)).length);
+    final int maxDaysBack = maxDaysFront ~/ 2;
+
+    final List<Widget> daysFront = days.sublist(0, min(maxDaysFront, days.length));
+
+    List<Widget> daysBack = <Widget>[];
+    if (maxDaysFront + maxDaysBack < days.length) {
+      daysBack = days.sublist(days.length - maxDaysBack);
+    }
+
+    return Column(
+      children: <Widget>[
+        Wrap(spacing: 4, runSpacing: 4, children: daysFront),
+        if (daysBack.isNotEmpty) ...<Widget>[
+          const Padding(padding: EdgeInsets.symmetric(vertical: 5), child: Icon(Icons.more_vert, size: 42)),
+          Wrap(spacing: 4, runSpacing: 4, children: daysBack),
+        ]
+      ],
+    );
+  }
+}
+
+class RecurrenceOptionsIndicatorDay extends StatelessWidget {
+  final DateTime day;
+  final RecurrenceOptionsIndicatorType type;
+
+  const RecurrenceOptionsIndicatorDay({super.key, required this.day, required this.type});
+
+  Color getBackgroundColor(BuildContext context) => type == RecurrenceOptionsIndicatorType.added
+      ? Theme.of(context).own().success
+      : type == RecurrenceOptionsIndicatorType.removed
+          ? Theme.of(context).colorScheme.error
+          : Theme.of(context).colorScheme.primary;
+
+  Color getForegroundColor(BuildContext context) => type == RecurrenceOptionsIndicatorType.added
+      ? Theme.of(context).own().onSuccess
+      : type == RecurrenceOptionsIndicatorType.removed
+          ? Theme.of(context).colorScheme.onError
+          : Theme.of(context).colorScheme.onPrimary;
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle dateStyle = TextStyle(color: getForegroundColor(context));
+    final TextPainter textPainter = TextPainter(
+      text: TextSpan(text: localeManager.formatDate(DateTime(2022, 12, 12)), style: dateStyle),
```

Comment:
funny... Maybe add a comment here to make sure people understand what this is doing

Also, maybe there are languages where this is not the longest, but I don't care...